src/System.Management.Automation/engine/pipeline.cs

```diff
@@ -412,6 +401,9 @@ internal int AddCommand(CommandProcessorBase commandProcessor, int readFromComma
 
             _commands.Add(commandProcessor);
 
+            // We will log event(s) about the pipeline execution details if any command in the pipeline requests that.
+            _needToLog |= commandProcessor.CommandRuntime.LogPipelineExecutionDetail;
```

Comment:
We could do `_eventLogBuffer ??= commandProcessor.CommandRuntime.LogPipelineExecutionDetail ? new List<string>() : null;` and remove `_needToLog` at all.

Reply:
About where to initialize the buffer list, see my comment above.

Reply:
Also, I think your suggestion is way more obscure, which will impair readability.